integrations/opensearch/src/haystack_integrations/components/retrievers/opensearch/open_search_hybrid_retriever.py

```diff
@@ -0,0 +1,347 @@
+# SPDX-FileCopyrightText: 2023-present deepset GmbH <info@deepset.ai>
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from typing import Any, Dict, List, Literal, Optional, Union
+
+from haystack import Pipeline, default_from_dict, default_to_dict, logging
+from haystack.components.embedders import SentenceTransformersTextEmbedder
+from haystack.components.joiners import DocumentJoiner
+from haystack.components.joiners.document_joiner import JoinMode
+from haystack.document_stores.types import FilterPolicy
+from haystack.lazy_imports import LazyImport
+from haystack.utils import ComponentDevice, Secret
+
+from haystack_integrations.components.retrievers.opensearch import OpenSearchBM25Retriever, OpenSearchEmbeddingRetriever
+from haystack_integrations.document_stores.opensearch import OpenSearchDocumentStore
+
+logger = logging.getLogger(__name__)
+
+# Use LazyImport to conditionally import haystack super_component, which is only available in newer versions
+with LazyImport(
+    "To use the OpenSearchHybridRetriever you need a more recent version of haystack. Run 'pip install haystack-ai>=2.13.0'"  # noqa: E501
+) as haystack_imports:
+    from haystack import super_component
+
+# Trigger an error message when the OpenSearchHybridRetriever is imported without haystack-ai>=2.13.0
+haystack_imports.check()
+
+
+@super_component
+class OpenSearchHybridRetriever:
+    """
+    A hybrid retriever that combines embedding-based and keyword-based retrieval from OpenSearch.
+
+    This component requires haystack-ai>=2.13.0 to work properly.
+    """
+
+    def __init__(
+        self,
+        document_store: OpenSearchDocumentStore,
+        *,
+        # SentenceTransformersTextEmbedder
+        model: str = "sentence-transformers/all-mpnet-base-v2",
```

Comment:
Sorry to derail this a bit, but I'm unsure if we should hardcode the TextEmbedder users have to use in this SupreComponent. For example, it would make it unusable for the majority of deepset Platforms use cases since we use a custom text embedder in all of our templates. 

I'd prefer if we allow a user to pass in a fully initialized Text Embedder as init param so we don't restrict users into having to use SentenceTransformers.

Reply:
This should also hopefully reduce the init signature and simplify the sede by not needing to support all the params of the TextEmbedder

Reply:
I understand that makes sense - I briefly discussed it with Julian, and we agreed to use the SentenceTransformer as it was the most popular one. 

I opened a ticket here: https://github.com/deepset-ai/haystack/issues/9351

Reply:
Thanks for creating the issue! I added one bullet point to the issue. I admit that I didn't take into account that multiple TextEmbedders could be supported easily with a TextEmbedder protocol. As it simplifies the HybridRetriever and will allow more users to benefit from it, I believe we should add the protocol before continuing with the HybridRetriever. @davidsbatista We can talk about later today.